src/node_messaging.cc

```diff
@@ -98,8 +103,15 @@ class DeserializerDelegate : public ValueDeserializer::Delegate {
     uint32_t id;
     if (!deserializer->ReadUint32(&id))
       return MaybeLocal<Object>();
-    CHECK_LT(id, host_objects_.size());
-    return host_objects_[id]->object(isolate);
+    if (id != kNormalObject) {
+      CHECK_LT(id, host_objects_.size());
+      return host_objects_[id]->object(isolate);
+    }
+    Local<Value> object;
+    if (!deserializer->ReadValue(env_->context()).ToLocal(&object))
```

Comment:
Shouldn't this be 
```suggestion
    if (!deserializer->ReadValue(isolate->GetCurrentContext()).ToLocal(&object))
```

anyway? Deserialization is [*not* guaranteed to happen in the main context](https://nodejs.org/api/worker_threads.html#workermovemessageporttocontextport-contextifiedsandbox), and as a nice little advantage adding `env_` becomes unnecessary.

Reply:
I thought about this for a bit.

`isolate->GetCurrentContext()` has the same issue you point out with `env_->context()` (may not be the context that's being deserialized into) but `ReadValue()` doesn't use it anyway so it doesn't really matter.

I'll use `isolate->GetCurrentContext()` but an empty `Local<Context>` would work too.

Reply:
Yeah, makes sense. We do enter the right context here: https://github.com/nodejs/node/blob/7816dad06006f5ef856877e6274055f1257c34d9/src/node_messaging.cc#L144